app/views/incomeSources/manage/ManageObligations.scala.html

```diff
@@ -0,0 +1,204 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import _root_.implicits.ImplicitDateFormatterImpl
+@import views.html.layouts.unifiedLayout
+@import views.html.components._
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import models.incomeSourceDetails.viewmodels.ObligationsViewModel
+
+@this(
+        mainTemplate: unifiedLayout,
+        implicitDateFormatter: ImplicitDateFormatterImpl,
+        continueButton: ContinueButton,
+        formWithCSRF: FormWithCSRF,
+        h1: h1,
+        h2: h2,
+        p: p,
+        link: link,
+        hr: hr,
+        govukTable: GovukTable
+)
+
+
+@(
+        sources: ObligationsViewModel,
+        mode: String,
+        businessName: String,
+        taxYear: String,
+        changeTo: String,
+        isAgent: Boolean,
+        backUrl: String,
+        postAction: Call
+)(implicit messages: Messages, user: auth.MtdItUser[_])
+
+@import implicitDateFormatter.longDate
+
+@getMessage(key: String, args: String*) = @{
+    messages(s"business-added.$key", args: _*)
+}
+
+@getUniqueMessage(key: String, args: String*) = @{
+    messages(s"incomeSources.add.manageObligations.$key", args: _*)
+}
+
+@getBusinessName(name: String, mode: String) = @{
+    if(name == "Not Found") getUniqueMessage("sole-trader")
+    else name
+}
+
+@getPeriodMessage(changeTo: String) = @{
+    if (changeTo == "annual") getUniqueMessage("annually")
+    else getUniqueMessage("quarterly")
+}
+
+@getTaxYears(taxYear:String) = @{
+    taxYear.subSequence(0,4) + " " + getUniqueMessage("to") + " " + taxYear.subSequence(5,9)
+}
+
+@getPageTitle = @{
+    val title = getUniqueMessage("h1")+" "+getBusinessName(businessName, mode)+" "+getUniqueMessage("h2")+" "+getPeriodMessage(changeTo)+" "+getUniqueMessage("tax-year")+" "+getTaxYears(taxYear)
+    title.trim
+}
+
+@mainTemplate(pageTitle = getPageTitle,
+    backUrl = Some(backUrl),
+    isAgent = isAgent,
+    btaNavPartial = user.btaNavPartial) {
+
+    @formWithCSRF(action = postAction) {
+
+        <div class="govuk-panel govuk-panel--confirmation" id="banner">
+            <div class="govuk-panel__body govuk-!-margin-bottom-3">
+                @getUniqueMessage("h1")
+            </div>
+            <h1 class="govuk-panel__title">
+                @getBusinessName(businessName, mode)
+            </h1>
+            <div class="govuk-panel__body">
+                @getUniqueMessage("h2")
+                @getPeriodMessage(changeTo)
+                @getUniqueMessage("tax-year")
+                @getTaxYears(taxYear)
+            </div>
+        </div>
+
+        <h2 class="govuk-label-wrapper">
+            <label class="govuk-label govuk-label--l govuk-!-margin-top-8">
+                @getUniqueMessage("heading")
+            </label>
+        </h2>
+
+        @if(sources.quarterlyObligationsDatesYearOne.nonEmpty) {
```

Comment:
this might be really better to use .forEach instead here, 
usage of .get or .head is really discouraged, this make code less flexible

Reply:
I need to check if it's empty to see if I need to render the table at all. Displaying the heads of the table depends on whether there is any data, it isn't done for every piece of data